### PR TITLE
update occ files:repair-tree command to also handle cases where the parent is -1 when it shouldn't be

### PR DESCRIPTION
**Pull request by icewind1991**
items should only have a parent of `-1` when the path is `''`. This adds some logic to determine the proper parent id to repair the invalid items.

### CODE REVIEW

**Review by ArtificialOwl — APPROVED** [review]
tested

File: apps/files/lib/Command/RepairTree.php [hunk 1]
```diff
@@ -51,9 +54,48 @@ protected function configure() {
...
+
+		foreach ($rows as $row) {
+			$output->writeln("Parent of file <info>{$row['fileid']}</info> (<info>{$row['path']}</info>) is <info>-1</info> but should be <info>{$row['parent_id']}</info> based on its path", OutputInterface::VERBOSITY_VERBOSE);
+
+			if ($fix) {
+				$query->setParameters([
+					'fileid' => $row['fileid'],
+					'parent' => $row['parent_id'],
+				]);
+				$query->execute();
```
Comment (come-nc) [inline apps/files/lib/Command/RepairTree.php:87]:
```suggestion
				$query->executeStatement();
```

File: apps/files/lib/Command/RepairTree.php [hunk 2]
```diff
@@ -51,9 +54,48 @@ protected function configure() {
...
+		$output->writeln("Found " . count($rows) . " file entries with an invalid parent");
+
+		if ($fix) {
+			$this->connection->beginTransaction();
+		}
+
+		$query = $this->connection->getQueryBuilder();
+		$query->update('filecache')
+			->set('parent', $query->createParameter('parent'))
+			->where($query->expr()->eq('fileid', $query->createParameter('fileid')));
```
Comment (come-nc) [inline apps/files/lib/Command/RepairTree.php:77]:
```suggestion
			$query = $this->connection->getQueryBuilder();
			$query->update('filecache')
				->set('parent', $query->createParameter('parent'))
				->where($query->expr()->eq('fileid', $query->createParameter('fileid')));
		}
```
The query is only used for fixing, no?

File: apps/files/lib/Command/RepairTree.php [hunk 3]
```diff
@@ -51,9 +54,48 @@ protected function configure() {
...
+				$query->execute();
+			}
+		}
+
+		if ($fix) {
+			$this->connection->commit();
+		}
+	}
+
+	private function repairPaths(bool $fix, OutputInterface $output) {
```
Comment (come-nc) [inline apps/files/lib/Command/RepairTree.php:96]:
```suggestion
	private function repairPaths(bool $fix, OutputInterface $output): void {
```

File: apps/files/lib/Command/RepairTree.php [hunk 4]
```diff
@@ -127,7 +167,49 @@ private function findBrokenTreeBits(): array {
 					$query->expr()->neq('f.path', 'f.name')
 				),
 				$query->expr()->neq('f.storage', 'p.storage')
-			));
+			))
+			->andWhere($query->expr()->neq('f.parent', $query->createNamedParameter(-1, IQueryBuilder::PARAM_INT)));
+
+		return $query->execute()->fetchAll();
```
Comment (come-nc) [inline apps/files/lib/Command/RepairTree.php:173]:
```suggestion
		return $query->executeQuery()->fetchAll();
```

File: apps/files/lib/Command/RepairTree.php [hunk 5]
```diff
@@ -127,7 +167,49 @@ private function findBrokenTreeBits(): array {
...
+						)
+
+					))
+				),
+				$query->expr()->eq('f.storage', 'p.storage')
+			))
+			->where($query->expr()->neq('f.path_hash', $query->createNamedParameter(md5(''))))
+			->andWhere($query->expr()->eq('f.parent', $query->createNamedParameter(-1, IQueryBuilder::PARAM_INT)));
 
 		return $query->execute()->fetchAll();
```
Comment (come-nc) [inline apps/files/lib/Command/RepairTree.php:214]:
```suggestion
		return $query->executeQuery()->fetchAll();
```

**Pull request closed without merging**